torch/distributions/truncated_normal.py

```diff
@@ -0,0 +1,100 @@
+import torch
+from torch.distributions.normal import Normal
+from torch.distributions.truncated_distribution import TruncatedDistribution
+import math
+
+
+def _standard_truncnorm_sample(lower_bound, upper_bound, sample_shape=torch.Size()):
```

Comment:
IIUC this is not reparameterized, and is only suitable as an implementation of `.sample()` (in particular it cannot differentiate wrt the bounds). I believe we can wrap this in a `torch.autograd.Function` and use the `erfinv` backward for gradients. Then it would also be suitable for `.rsample()`.